data_safe_haven/config/shm_config.py

```diff
@@ -27,10 +28,14 @@ def from_args(
     ) -> SHMConfig:
         """Construct an SHMConfig from arguments."""
         azure_sdk = AzureSdk(subscription_name=context.subscription_name)
-        admin_group_id = (
-            azure_sdk.entra_directory.get_id_from_groupname(context.admin_group_name)
-            or "admin-group-id-not-found"
-        )
+        try:
+            admin_group_id = azure_sdk.entra_directory.get_id_from_groupname(
+                context.admin_group_name
+            )
+        except DataSafeHavenMicrosoftGraphError as exc:
+            msg = f"Failed to get ID for group '{context.admin_group_name}'"
+            raise DataSafeHavenMicrosoftGraphError(msg) from exc
+
```

Comment:
How about this instead of adding a new method?

```suggestion
        if not (admin_group_id := azure_sdk.entra_directory.get_id_from_groupname(
            context.admin_group_name
        )):
            msg = f"Failed to get ID for group '{context.admin_group_name}'"
            raise DataSafeHavenMicrosoftGraphError(msg) from exc

```

Reply:
Won't this raise the exception if `admin_group_id` is `None`? `not None is True`

Reply:
Yes, if it's none (the lookup failed) or empty string (the lookup returned empty string, which shouldn't be possible). That's what we want, right?